message.go

```diff
@@ -338,8 +346,14 @@ func (m *Message) nextFrameToSend(maxSize int) ([]byte, frameFlags) {
 		var writer io.WriteCloser
 		m.encoder, writer = io.Pipe()
 		go func() {
-			m.WriteTo(writer)
-			writer.Close()
+			defer func() {
+				if p := recover(); p != nil {
+					log.Printf("PANIC in BLIP nextFrameToSend: %v\n%s", p, debug.Stack())
+				}
+			}()
+
+			_ = m.WriteTo(writer)
+			_ = writer.Close()
```

Comment:
Move writer.Close() to defer (and ensure it runs after recover() handling)

Add logging for errors returned by WriteTo or writer.Close while we're here.